packages/language-server/src/plugins/typescript/utils.ts

```diff
@@ -178,10 +178,13 @@ export function isVirtualFilePath(filePath: string) {
 }
 
 export function toVirtualAstroFilePath(filePath: string) {
-  if (isVirtualFrameworkFilePath('astro', filePath)) {
+  if (isVirtualAstroFilePath(filePath)) {
+    return filePath;
+  } else if(isAstroFilePath(filePath)) {
```

Comment:
This is what fixes perf, this else if was missing before and causing TS to search for files for a long time.